### PR TITLE
actionagent callback stderr if match regexp

### PR DESCRIPTION
**Pull request by chengjoey**
#### What type of this PR

Add one of the following kinds:
/kind bug

#### What this PR does / why we need it:
actionagent callback stderr if match regexp in every line

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda-org.erda.cloud/erda/workBench/projects/387/issues/all?id=70728&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIl0sImFzc2lnbmVlSURzIjpbIjEwMDEyMDUiXX0%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=431&type=TASK)

### CODE REVIEW

**Comment by sfwn** [comment]
Give me a screenshot about it.

File: modules/actionagent/file_watch.go [hunk 1]
```diff
@@ -146,3 +152,18 @@ func getMetaFromLogLine(line string) (bool, string, string) {
 	}
 	return true, k, v
 }
+
+func matchStdErrLine(regexpList []string, line string) bool {
+	var matched bool
+	defer func() {
+		_ = recover()
```
Comment (sfwn) [inline modules/actionagent/file_watch.go:159]: log when recover

File: modules/actionagent/file_watch.go [hunk 2]
```diff
@@ -146,3 +152,18 @@ func getMetaFromLogLine(line string) (bool, string, string) {
...
+	}()
+	for i := range regexpList {
+		reg := regexp.MustCompile(regexpList[i])
```
Comment (sfwn) [inline modules/actionagent/file_watch.go:162]: Compile all regexps once at very beginning.

File: modules/actionagent/step_prepare.go [hunk 3]
```diff
@@ -87,6 +92,10 @@ func (agent *Agent) prepare() {
 	} else {
 		agent.EasyUse.RunMultiStderr = f
 	}
+
+	// 5. set stderr regexp list
+	envStdErrRegexpStr := os.Getenv(EnvStdErrRegexpList)
+	_ = json.Unmarshal([]byte(envStdErrRegexpStr), &agent.StdErrRegexpList)
```
Comment (sfwn) [inline modules/actionagent/step_prepare.go:98]: handler errors correctly.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Comment by chengjoey** [comment]
> Give me a screenshot about it.

pipeline.yml:
image

result:
image

File: build/dockerfiles/Dockerfile-action-agent [hunk 4]
```diff
@@ -0,0 +1,21 @@
+FROM registry.cn-hangzhou.aliyuncs.com/terminus/terminus-golang:1.14 AS compile
```
Comment (sfwn) [inline build/dockerfiles/Dockerfile-action-agent:1]: Update base image as others.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**